eclair-core/src/main/scala/fr/acinq/eclair/api/Service.scala

```diff
@@ -102,7 +102,12 @@ trait Service extends Logging {
                 case JsonRPCBody(_, _, "send", JString(paymentRequest) :: Nil) =>
                   for {
                     req <- Future(PaymentRequest.read(paymentRequest))
-                    res <- (paymentInitiator ? SendPayment(req.amount.getOrElse(throw new RuntimeException("request without amounts are not supported")).amount, req.paymentHash, req.nodeId)).mapTo[PaymentResult]
+                    res <- (paymentInitiator ? SendPayment(req.amount.getOrElse(throw new RuntimeException("for requests without amount use send (amountMsat, paymentRequet)")).amount, req.paymentHash, req.nodeId)).mapTo[PaymentResult]
```

Comment:
Perhaps one branch could be dropped since once `JInt(amt)` is present we don't care about original amount?
```
for {
  req <- Future(PaymentRequest.read(paymentRequest))
  amount = (req.amount, rest) match {
    case (Some(amt), Nil) => amt.amount
    case (_, JInt(amt) :: Nil) => amt.toLong // overriding payment request amount with the one provided
    case (None, Nil) => throw new RuntimeException("you need to manually specify an amount for this payment request")
  }
  res <- (paymentInitiator ? SendPayment(amount, req.paymentHash, req.nodeId)).mapTo[PaymentResult]
} yield res
```